llm_studio/src/datasets/text_causal_language_modeling_ds.py

```diff
@@ -62,7 +62,9 @@ def __init__(self, df: pd.DataFrame, cfg: Any, mode: str = "train"):
 
         self.tokenizer = get_tokenizer(cfg)
 
-        self.prompts = get_texts(df, self.cfg, separator="")
+        self.raw_prompts = get_texts(df, self.cfg, separator="")
+        self.prompts = [self.parse_prompt(cfg, prompt) for prompt in self.raw_prompts]
```

Comment:
It is possible to move `.parse_prompt` to the `_get_sample` method and to the example logging.
By this, one does not need to keep two versions of the prompts.

Reply:
we use `parse_prompt` in the chat

Reply:
Ah sorry, I mistyped my idea:
What I meant was to remove `self.prompts` (and rename `self.raw_prompts` to `self.prompts`) and call `self.parse_prompt` in the `_get_sample` method and when logging an example.